UI/MainForm.cs

```diff
@@ -2016,22 +2016,33 @@ private void _checkForOrphanedSerials(bool showNotNecessaryMessage)
             {
                 var allConfigItems = execManager.Project.ConfigFiles.Select(file => file.ConfigItems).ToList();
                 OrphanedSerialsDialog opd = new OrphanedSerialsDialog(serials, allConfigItems);
+
+                void UpdateProject()
+                {
+                    ProjectHasUnsavedChanges = opd.HasChanged();
+                    var updatedConfigs = opd.GetUpdatedConfigs();
+
+                    for (int i = 0; i < execManager.Project.ConfigFiles.Count; i++)
+                    {
+                        execManager.Project.ConfigFiles[i].ConfigItems = updatedConfigs[i];
+                    }
+
+                    MessageExchange.Instance.Publish(execManager.Project);
+                }
+                
                 opd.StartPosition = FormStartPosition.CenterParent;
                 if (opd.HasOrphanedSerials())
                 {
-                    if (opd.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+                    if (opd.ShowDialog() == DialogResult.OK)
                     {
-                        ProjectHasUnsavedChanges = opd.HasChanged();
-                        var udpatedConfigs = opd.GetUpdatedConfigs();
-
-                        for (int i = 0; i < execManager.Project.ConfigFiles.Count; i++)
-                        {
-                            execManager.Project.ConfigFiles[i].ConfigItems = udpatedConfigs[i];
-                        }
-
-                        MessageExchange.Instance.Publish(execManager.Project);
+                        UpdateProject();
                     }
                 }
+                else if (opd.HasChanged())
+                {
+                    // If there are no orphaned serials, serials can still be changed automatically
+                    UpdateProject();
```

Comment:
We could add a `TimeoutMessageDialog` here, but I think these "serials" are an implementation detail we don't need to expose to users.